src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -176,10 +215,29 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
             bool remapPorts = true;
 
-            // If type is still null at this point return a dummy node
             if (type == null)
             {
-                node = CreateDummyNode(obj, typeName, assemblyName, functionName, inPorts, outPorts);
+                if (typeName == "CoreNodeModels.Formula")
+                {
+                    var code = obj["Formula"].Value<string>();
+                    var formulaConverter = new MigrateFormulaToDS();
+                    var convertedCode = formulaConverter.ConvertFormulaToDS(code);
+
+                    node = DeserializeAsCBN(convertedCode + ";", obj, guid);
+                    if (code == convertedCode)
```

Comment:
is it possible that the code is exactly the same and the conversion is successful?

Reply:
Good point, I'll try to move this warning inside the `catch` itself.